EmberBSD local test harness repair, AI-assisted; not submitted upstream.
Pass flags separately, pair input/output modules and propagate failures.

--- tests/aslts/Makefile.def.orig
+++ tests/aslts/Makefile.def
@@ -25,38 +25,38 @@
 ifndef ASLTS_AMLDIR
 ifeq ("$(ASLTS_MODE)", "n32")
 	ASLTS_AMLDIR=nopt/32
-	ASLTS_ASLFLAGS="-oa -r 1"
+	ASLTS_ASLFLAGS=-oa -r 1
 endif
 ifeq ("$(ASLTS_MODE)", "n64")
 	ASLTS_AMLDIR=nopt/64
-	ASLTS_ASLFLAGS="-oa -r 2"
+	ASLTS_ASLFLAGS=-oa -r 2
 endif
 ifeq ("$(ASLTS_MODE)", "o32")
 	ASLTS_AMLDIR=opt/32
-	ASLTS_ASLFLAGS="-r 1"
+	ASLTS_ASLFLAGS=-r 1
 endif
 ifeq ("$(ASLTS_MODE)", "o64")
 	ASLTS_AMLDIR=opt/64
-	ASLTS_ASLFLAGS="-r 2"
+	ASLTS_ASLFLAGS=-r 2
 endif
 ifeq ("$(ASLTS_MODE)", "aslminus")
 	ASLTS_AMLDIR=aslminus/
-	ASLTS_ASLFLAGS="-oa -r 2"
+	ASLTS_ASLFLAGS=-oa -r 2
 endif
 ifeq ("$(ASLTS_MODE)", "aslplus")
 	ASLTS_AMLDIR=aslplus/
-	ASLTS_ASLFLAGS="-oa -r 2"
+	ASLTS_ASLFLAGS=-oa -r 2
 endif
 ifeq ("$(ASLTS_MODE)", "aslconvert")
 	ASLTS_AMLDIR=aslconvert/
-	ASLTS_ASLFLAGS="-oa -r 2"
+	ASLTS_ASLFLAGS=-oa -r 2
 endif
 
 endif
 ASLTS_VER ?= $(ASLTS_VER)
 
-COMMON_NO_OUTPUT_ASL_FLAGS= "-of -cr -vs"
-COMMON_ASL_FLAGS= $(COMMON_NO_OUTPUT_ASL_FLAGS)" -l -sc -sa -ic -ta -ts -so -lm -ln -ls -li"
+COMMON_NO_OUTPUT_ASL_FLAGS= -oE -of -cr -vs
+COMMON_ASL_FLAGS= $(COMMON_NO_OUTPUT_ASL_FLAGS) -l -sc -sa -ic -ta -ts -so -lm -ln -ls -li
 
 COMPILER_LOG="$(TOP)/tmp/aml/$(ASLTS_VER)/compile.txt"
 COMPILER_ERROR_LOG="$(TOP)/tmp/aml/$(ASLTS_VER)/error.txt"
@@ -95,7 +95,7 @@
 			echo "---- Compile: $$j.asl" >> $(COMPILER_LOG); \
 			>&2 printf "%-18s" "    Type: $(ASLTS_AMLDIR) "; \
 			>&2 printf  "Compile"; \
-			"$(ASL)" $(ASLTS_ASLFLAGS) "$(COMMON_ASL_FLAGS)" $(ADD_ASLFLAGS) $$j.asl >> $(COMPILER_LOG) 2>> $(COMPILER_ERROR_LOG); \
+			"$(ASL)" $(ASLTS_ASLFLAGS) $(COMMON_ASL_FLAGS) $(ADD_ASLFLAGS) $$j.asl >> $(COMPILER_LOG) 2>> $(COMPILER_ERROR_LOG); \
 			ret=$$?; \
 			echo "" >> $(COMPILER_LOG); \
 			>&2 printf " => Removing files"; \
@@ -127,7 +127,8 @@
 	else \
 		echo "Bad iASL 0: <$(ASL)> does not exist"; \
 		exit 1; \
-	fi;
+	fi; \
+	exit $$rval
 
 # Make-install one particular Test Case for all modes.
 
@@ -146,12 +147,14 @@
 	fi
 
 
-TEST_SEQUENCE_ASL_FLAGS="$(ASLTS_ASLFLAGS) $(COMMON_NO_OUTPUT_ASL_FLAGS)"
+TEST_SEQUENCE_ASL_FLAGS=$(ASLTS_ASLFLAGS) $(COMMON_NO_OUTPUT_ASL_FLAGS)
 
 disasm_test_compile: $(TOP)/tmp/aml/$(ASLTS_VER)/$(ASLTS_AMLDIR)
 	@rval=0; \
+	set -- ${AMLMOD}; \
 	for j in ${ASLMOD} $(ASLMODADD); do \
-		for k in ${AMLMOD}; do \
+		if [ $$# -eq 0 ]; then rval=2; break; fi; \
+		k=$$1; shift; \
 			>&2 printf  " => Compile with externals in place"; \
 			"$(ASL)" -p $$k-extInPlace -oE $(ADD_ASLFLAGS) $(TEST_SEQUENCE_ASL_FLAGS) $$j.asl >> $(COMPILER_LOG) 2>> $(COMPILER_ERROR_LOG); \
 			ret=$$?; \
@@ -160,8 +163,8 @@
 				>&2 printf " [[ Error: compiling $$k.aml with externals in place failed ]]\n"; \
 				>&2 printf "          Flags used: -p $$k-extInPlace -oE $(ADD_ASLFLAGS) $(TEST_SEQUENCE_ASL_FLAGS)\n\n"; \
 			fi; \
-		done; \
-	done; \
+	done; \
+	if [ $$# -ne 0 ]; then rval=2; fi; \
 	if [ $$rval != 0 ]; then exit 1; fi
 
 disasm_test_disassemble: disasm_test_compile
@@ -195,10 +198,11 @@
 		ret=$$?; \
 		if [ $$ret != 0 ]; then rval=2; echo "**** mv failed!" >> $(COMPILER_LOG) 2>> $(COMPILER_ERROR_LOG); break; fi; \
 	done; \
+	exit $$rval
 
 install_disasm_test_case:
 	if [ "x$(ADD_ASLFLAGS)" != "x-f" ]; then \
-		make disasm_test_recompile; \
+		make disasm_test_recompile || exit $$?; \
 		make clean; \
 	else \
 		>&2 echo "Skipping disassembler test for this test case"; \
@@ -220,8 +224,10 @@
 
 converter_test_recompile: converter_test_convert
 	@rval=0; \
+	set -- ${AMLMOD}; \
 	for j in ${ASLMOD} $(ASLMODADD); do \
-		for k in ${AMLMOD}; do \
+		if [ $$# -eq 0 ]; then rval=2; break; fi; \
+		k=$$1; shift; \
 			>&2 printf " => Recompile"; \
 			"$(ASL)" $(ADD_ASLFLAGS) $(TEST_SEQUENCE_ASL_FLAGS) -p $$k $$j.dsl >> $(COMPILER_LOG) 2>> $(COMPILER_ERROR_LOG); \
 			ret=$$?; \
@@ -236,12 +242,13 @@
 			mv $$k.aml $(TOP)/tmp/aml/$(ASLTS_VER)/$(ASLTS_AMLDIR); \
 			ret=$$?; \
 			if [ $$ret != 0 ]; then rval=2; echo "**** mv failed!" >> $(COMPILER_LOG) 2>> $(COMPILER_ERROR_LOG); break; fi; \
-		done; \
-	done; \
+	done; \
+	if [ $$# -ne 0 ]; then rval=2; fi; \
+	exit $$rval
 
 install_converter_test_case:
 	if [ "x$(ADD_ASLFLAGS)" != "x-f" ]; then \
-		make converter_test_recompile; \
+		make converter_test_recompile || exit $$?; \
 		make clean; \
 	else \
 		>&2 echo "Skipping converter test for this test case"; \
